tensorrt_llm/_torch/pyexecutor/sampler/beam_search.py

```
@@ -1210,6 +1216,41 @@ def _builder() -> BeamHistory | None:
+    """
+    assert token_tensor.dim() == 3 and logprobs_tensor.dim() == 3, (
+        f"Token and logprobs tensors must have 3 dimensions (beam_width, num_tokens, num_logprobs). \
+        Got shapes (token_tensor) {token_tensor.shape} and (logprobs_tensor) {logprobs_tensor.shape} instead"
+    )
+
+    token_log_probs: list[list[dict[int, Logprob]]] = []
+    token_list = token_tensor.tolist()
+    logprobs_list = logprobs_tensor.tolist()
+    for beam_idx in range(token_tensor.shape[0]):
+        beam_token_log_probs: list[dict[int, Logprob]] = []
+        for topk_token, topk_logprob in zip(token_list[beam_idx], logprobs_list[beam_idx]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)